src/client/metadataTransfer.ts

```diff
@@ -115,14 +115,26 @@ export abstract class MetadataTransfer<
       }
       return result;
     } catch (e) {
-      const err = e as Error;
+      const err = e as Error | SfError;
       const error = new SfError(messages.getMessage('md_request_fail', [err.message]), 'MetadataTransferError');
-      error.setData({
-        id: this.id,
-      });
+
       if (error.stack && err.stack) {
```

Comment:
how would `error.stack` ever be truthy if we just constructed it?

Reply:
all errors have a stack trace set when instantiated:
```
node -e  'const err= new Error(); console.log(err.stack);'
```

<img width="1290" alt="Screenshot 2024-04-03 at 6 23 26 PM" src="https://github.com/forcedotcom/source-deploy-retrieve/assets/6853656/efcfbe88-3c90-452b-8136-b78df97c1937">